core/src/main/scala/org/apache/spark/metrics/sink/opentelemetry/OpenTelemetryPushReporter.scala

```diff
@@ -0,0 +1,360 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.metrics.sink.opentelemetry
+
+import java.nio.file.{Files, Paths}
+import java.util.{Locale, SortedMap}
+import java.util.concurrent.TimeUnit
+
+import com.codahale.metrics._
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.metrics.{DoubleGauge, DoubleHistogram, LongCounter}
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator
+import io.opentelemetry.context.propagation.ContextPropagators
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter
+import io.opentelemetry.sdk.OpenTelemetrySdk
+import io.opentelemetry.sdk.metrics.SdkMeterProvider
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader
+import io.opentelemetry.sdk.resources.Resource
+
+private[spark] class OpenTelemetryPushReporter(
+    registry: MetricRegistry,
+    pollInterval: Int = 10,
+    pollUnit: TimeUnit = TimeUnit.SECONDS,
+    host: String = "http://localhost",
+    port: String = "4317",
```

Comment:
@dongjoon-hyun 4317 is port for grpc and 4318 is for HTTP/HTTPS. Should I rename the port to `grpcPort`?

https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/

Reply:
To @TQJADE , you are using `http://` in line 39, aren't you? Are you using `grpc` instead of HTTP here?

Reply:
https://github.com/open-telemetry/opentelemetry-java/blob/7522bfe141524d52649b5c344c643104b57af608/exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterBuilderUtil.java#L32-L47
Here there is a endpoint validation, it required the endpoint to start with http/https

Reply:
What does that mean? Do you mean `OpenTelemetry` requires `HTTP` with 4317 ? If we need to use `4317` with HTTP, which HTTP call use 4318 port?

Reply:
Here I am using grpc endpoint, when I tested, the endpoint should be started with `http` or `https`. The default value of Opentelmetry GRPC endpoint port is 4317. HTTP endpoint port is 4318.

Reply:
It seems that we are repeating the same questions and answer again and again. :) 

What I asked to confirm is that why we are not using the default value of HTTP endpoint port (4318) here when the endpoint starts with `http`. This PR uses `http://localhost:4317` instead of `http://localhost:4318`, doesn't it? We know that your sentence is correct, but it sounds like contradiction. We need an explanation for this contradiction.
>  the endpoint should be started with http or https. The default value of Opentelmetry GRPC endpoint port is 4317. HTTP endpoint port is 4318.